tests/core/utils/test_block.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+import rlp
+
+from evm.utils.blocks import (
+    add_block_number_to_hash_lookup,
+    get_score,
+    persist_block_to_db,
+    get_block_header_by_hash,
+    lookup_block_hash,
+)
+
+from evm.db import (
+    get_db_backend,
+)
+
+from evm.vm.flavors.frontier.blocks import (
+    FrontierBlock,
+)
+
+from evm.rlp.headers import (
+    BlockHeader,
+)
+
+from evm.utils.db import (
+    make_block_hash_to_score_lookup_key,
+    make_block_number_to_hash_lookup_key,
+)
+
+db = get_db_backend()
+header = BlockHeader(1,1,1)
+block = FrontierBlock(header, db)
+
+def test_add_block_number_to_hash_lookup():
+    block_number_to_hash_key = make_block_number_to_hash_lookup_key(1)
+    assert not db.exists(block_number_to_hash_key)
+    add_block_number_to_hash_lookup(db, block)
+    assert db.exists(block_number_to_hash_key)
+
+def test_perist_block_to_db():
+    block_to_hash_key = make_block_hash_to_score_lookup_key(block.hash)
+    assert not db.exists(block_to_hash_key)
+    persist_block_to_db(db, block)
+    assert db.exists(block_to_hash_key)
+
+def test_get_score():
+    block_to_hash_key = make_block_hash_to_score_lookup_key(block.hash)
+    score = rlp.decode(db.get(block_to_hash_key), sedes=rlp.sedes.big_endian_int)
+    assert get_score(db, block.hash) == score
+
+def test_get_block_header_by_hash():
+    block_header = get_block_header_by_hash(db, block.hash)
+    assert block_header == header
```

Comment:
There is a utility I wrote for this in exact purpose that I recommend using.

It's currently living in `evm.utils.fixture_tests` but really it should be moved to `evm.utils.rlp`.  It's called `assert_rlp_equal` and it both ensures that all fields are equal as well as printing out nice error messages when they are not.